### ISSUE TITLE
Quest order different from what I set

### ISSUE THREAD

**Issue by Helium314** [westnordost/StreetComplete#1759]
tl;dr:
When changing the quest order, after closing and opening the quest list, the quest order sometimes is not what I had just set.
This is likely related to issue #893, maybe the same.

longer version:
Sometimes I noticed that the quest order is different from what I remember, and also quests I had switched off are active. This is rather rare, I never was able to reproduce it and I never really had a big problem with this.

But recently I encountered that problem again in a more troublesome way:
I wanted to preload quests for a train trip, but wanted to preload only quests that are actually answerable from the train (e.g. building type, roof shape, road surface).
Since I wanted to restore my old quests again after preloading, I re-arranged the list so all my enabled quests are on top, and disabled ones are at the bottom. This way I just need to remember the lowest enabled quest instead of every single one.
I scrolled around on the map to load some tiles along the way and then went back to the quest screen. After disabling all quests and not finding most, but not all, of the what-is-the-surface quests on top the list, I noticed that the order was completely different from what I had previously set. I always keep all the surface quests together so I can quickly move the is-lit-quest above or below, depending on time of the day.
The list is not completely random, it appears that the previously enabled quests are still partially grouped together (in 3 or 4 groups instead of 1). But still, going back to the quest settings I previously had is quite a bit of work...

So please either fix this, or implement quest profiles (since this was an attempt to work around the non-existence of profiles)

I noted above that sometimes disabled quests are active again: I suspect that the same bug is causing this. If the internal quest order is different from the one that is shown on display I might en/disable a different quest that I actually tap on.

**Comment by westnordost** [comment]
The problem is known, what i need is a reliable (minimal) reproduction

**Comment by westnordost** [comment]
So, I am closing this bug as a duplicate of the unresolved #893 . Please put any information you find in that ticket
